vscode-smart-contracts/src/commands.ts

```diff
@@ -304,3 +319,59 @@ async function installCargoGenerate() {
     throw new Error("Failed installing cargo-generate");
   }
 }
+
+/**
+ * Generate JavaScript/TypeScript clients for the smart contract module.
+ */
+export async function generateJsClients() {
+  const editor = vscode.window.activeTextEditor;
+  if (editor === undefined) {
+    vscode.window.showErrorMessage(
+      "Unable to determine smart contract project. Open a file in the smart contract project that you want to build"
+    );
+    return;
+  }
+  const cwd = path.dirname(editor.document.uri.fsPath);
+  const projectDir = await locateCargoProjectDir(cwd);
+  const moduleFilePath = path.join(
+    projectDir,
+    DEFAULT_BUILD_OUT_DIR_NAME,
+    "module.wasm.v1"
+  );
+  const outDirPath = path.join(projectDir, DEFAULT_JS_GEN_OUT_DIR_NAME);
+  const workspaceFolder = vscode.workspace.getWorkspaceFolder(
+    editor.document.uri
+  );
+  if (!fs.existsSync(moduleFilePath)) {
+    const build_contract_action = "Build contract";
+    const action = await vscode.window.showErrorMessage(
+      `A compiled Wasm module could not be found. Please compile the smart contract via the "build contract" command. (Expect location: ${moduleFilePath})`,
+      build_contract_action
+    );
+    if (action !== build_contract_action) {
+      return;
+    }
+    const task = await buildTask("outDir");
+    if (task === undefined) {
+      return;
+    }
+    const exitCode = await executeAndAwaitTask(task);
+    if (exitCode !== 0) {
+      return;
```

Comment:
**Question:** Is the user shown anything in this case, and will they understand why it did not generate anything in this case?

Reply:
Good catch. I am now throwing errors there instead of silently failing.
The case I had tested with was when it couldn't determine which project to build, in which case the build task threw an error on its own.